### ISSUE TITLE
CPU and memory usage turns high when EM.run in rescue block

### ISSUE THREAD

**Issue by halida** [eventmachine/eventmachine#482]
here is the code:

```ruby
require 'eventmachine'

def run
  EventMachine.run do
    EM.next_tick{ puts "12" }
  end
end

def main
  begin
    raise
  rescue Exception
    run
  end
end

main
```

After run this code, memory usage gradually increase, and CPU usage is high.
I'm using gdb to check this out, found lots of Proc object created, and this Proc object is this one in run_deferred_callbacks :
https://github.com/eventmachine/eventmachine/blob/cfb1f71a35b1a10e5821bca9841fee3080ec1685/lib/eventmachine.rb#L975

My enviroment:
OSX 10.9,
rvm + ruby 2.0.0-p353
eventmachine (1.0.3)

**Comment by sodabrew** [comment]
This sounds a lot like #452. I wasn't able to reproduce so I abandoned #573, but I will take another look at it.

**Comment by sodabrew** [comment]
Resolved by #452
